### ISSUE TITLE
[google_maps_flutter] Convert to Pigeon

### ISSUE THREAD

**Issue by stuartmorgan-g** [flutter/flutter#117907]
Part of https://github.com/flutter/flutter/issues/117844

~This is a case where we may want to build more instance-tracking functionality into Pigeon itself before doing a conversion, since we currently use per-instance channels.~ This is now implemented.

## Progress

- [x] `google_maps_flutter_android`
  - [x] Convert inspector calls
  - [x] Convert Dart->Host calls at the method level
  - [x] Convert data objects passed in method calls (can be done incrementally)
    - [x] `PlatformCameraUpdate`
    - [x] `PlatformCircle`
    - [ ] `PlatformHeatmap` - https://github.com/flutter/packages/pull/10758
    - [x] `PlatformMarker`
    - [x] `PlatformPolygon`
    - [x] `PlatformPolyline`
    - [x] `PlatformTileOverlay`
  - [x] Convert Host->Dart calls - https://github.com/flutter/packages/pull/7040
  - [x] Convert platform view creation params ~(blocked on https://github.com/flutter/flutter/issues/150631)~ - https://github.com/flutter/packages/pull/7207
- [x] `google_maps_flutter_ios`
  - [x] Convert inspector calls - https://github.com/flutter/packages/pull/6937
  - [x] Convert Dart->Host calls at the method level - https://github.com/flutter/packages/pull/7079
  - [x] Convert data objects passed in method calls (can be done incrementally, and the Dart portion of the Android conversions of these classes can be re-used modulo any platform-SDK-specific differences)
    - [x] `PlatformCameraUpdate`
    - [x] `PlatformCircle`
    - [x] `PlatformHeatmap` - https://github.com/flutter/packages/pull/10754
    - [x] `PlatformMarker`
    - [x] `PlatformPolygon`
    - [x] `PlatformPolyline`
    - [x] `PlatformTileOverlay`
  - [x] Convert Host->Dart calls - https://github.com/flutter/packages/pull/7086
  - [x] Convert platform view creation params ~(blocked on https://github.com/flutter/flutter/issues/150631)~ - https://github.com/flutter/packages/pull/7207

**Comment by stuartmorgan** [comment]
This is a large enough API surface that it would almost certainly be best done incrementally, by converting parts of the API in individual PRs.

**Comment by stuartmorgan** [comment]
Heatmaps are all that's left now. Migrating those will let us clean up a fair number of JSON conversion utility methods on the native sides, but now that almost everything is converted there's much less chance of someone adding new code starting off following the wrong pattern by looking at the existing code, so that last step is low priority.
